config/src/parameters/user.rs

```diff
@@ -308,24 +295,22 @@ pub(crate) fn private_key_from_env<E: Error>(
 #[derive(Debug)]
 pub struct Genesis {
     pub public_key: PublicKey,
-    pub private_key: Option<PrivateKey>,
     pub file: Option<PathBuf>,
 }
 
 impl Genesis {
-    fn parse(self, cli: CliContext) -> Result<actual::Genesis, GenesisConfigError> {
-        match (self.private_key, self.file, cli.submit_genesis) {
-            (None, None, false) => Ok(actual::Genesis::Partial {
+    fn parse(self, _cli: CliContext) -> Result<actual::Genesis, GenesisConfigError> {
+        match self.file {
+            None => Ok(actual::Genesis::Partial {
```

Comment:
As far as I understand, the new design implies that every peer should submit the genesis?

If so, then there shouldn't be `Genesis::Partial`/`Genesis::Full`, it should be always complete.